lib/rules.js

```diff
@@ -184,6 +185,234 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
+                                         {type: 'Identifier', name: 'nthRoot'},
+                                         [nthRoot.args[0]]
+                                     ))
+                }
+                return applyRule(CANCEL_EXPONENT,
+                                 build.applyNode(
+                                     {type: 'Identifier', name: 'nthRoot'},
+                                     [nthRoot.args[0]]
+                                 ))
+            })
+        )
+        return result
+    }
+)
+
+export const primeFactorization = (number, result = []) => {
+    const root = Math.sqrt(number)
+
+    let x = 2
+
+    if (number % x) {
+        x = 3
+        while (number % x && x < root) {
+            x = x + 2
+        }
+    }
+
+    x = x <= root ? x : number
+
+    result.push(x)
+
+    return x === number
+        ? result
+        : primeFactorization(number / x, result)
+}
+
+// e.g. 12 -> 2 * 2 * 3
+export const FACTOR_INTO_PRIME = defineRule(
+    (node) => {
+        return query.isNumber(node) ? {node} : null
+    },
+
+    (node) => {
+        const factors = primeFactorization(query.getValue(node))
+        const result = build.applyNode(
+            'mul',
+            factors.map(arg => {
+                return build.numberNode(arg)
+            })
+        )
+        return result
+    }
+)
+
+// e.g. nthRoot(2 * 2 * 2, 2) -> nthRoot((2 * 2) * 2)
+export const GROUP_TERMS_BY_ROOT = defineRule(
+    (node) => {
+        let canGroupTerms = false
+        canGroupTerms = canApplyRule(DISTRIBUTE_NTH_ROOT, node)
+        return canGroupTerms ? {node} : null
+    },
+
+    (node) => {
+        const radicand = node.args[0]
+        const index = node.args[1]
+
+        // dictionary storing the number of times a constant appears
+        // e.g. 2 * 2 * 2 => {'2': 3} , 2 appears 3 times
+        const count = {}
+        radicand.args.forEach(arg => {
+            const value = query.getValue(arg)
+            count[value] ? count[value]++ : count[value] = 1
+        })
+
+        const flatten = arr => arr.reduce(
+            (acc, val) => acc.concat(
+                Array.isArray(val) ? flatten(val) : val
+            ),
+            []
+        )
+
+        const result = build.applyNode(
+            {type: 'Identifier', name: 'nthRoot'},
+            [build.applyNode(
+                'mul',
+                flatten(Object.keys(count).map(key => {
+                    let leftover = count[key]
+                    const times = query.getValue(index)
+
+                    const args = []
+
+                    while(leftover - times > 0) {
+                        leftover -= times
+                        args.push(Array(times).fill(build.numberNode(key)))
+                    }
+                    leftover == 1
+                        ? args.push(build.numberNode(key))
+                        : args.push(Array(leftover).fill(build.numberNode(key)))
```

Comment:
I think this last line should be
```
: args.push(...Array(leftover).fill(build.numberNode(key)))
```
It should be possible to remove the ternary and use:
```
args.push(...Array(leftover).fill(build.numberNode(key)))
```